### PR TITLE
Add Feature Manager and use it to manage dbconnect integration

### PR DESCRIPTION
**Pull request by kartikgupta-db**
## Changes
* Add Feature Manager interface to manage feature detection and feature flags. 
* Use Feature Manager to manage detection of dbconnect.

https://user-images.githubusercontent.com/88345179/228900439-8bf608f8-d9f4-489a-b89c-7daade8db5a1.mov

https://user-images.githubusercontent.com/88345179/228900444-8798609c-42e9-49d7-aba7-149c7de6b4ef.mov

### CODE REVIEW

File: packages/databricks-vscode/src/feature-manager/FeatureManager.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+export type FeatureEnableAction = () => Promise<void>;
+export interface FeatureState {
+    avaliable: boolean;
+    reason?: string;
+    action?: FeatureEnableAction;
+}
+
+export type FeatureId = "debugging.dbconnect";
+export const disabledFeatures: FeatureId[] = ["debugging.dbconnect"];
```
Comment (fjakobs) [inline packages/databricks-vscode/src/feature-manager/FeatureManager.ts:12]: I'd rather have `disabledFeatures` a constructor argument and pass it in from `extension.ts`

File: packages/databricks-vscode/src/extension.ts [hunk 2]
```diff
@@ -356,8 +376,45 @@ export async function activate(
...
+        "debugging.dbconnect",
+        dbConnectAccessVerifier
+    );
+    context.subscriptions.push(
+        dbConnectAccessVerifier,
+        featureManager.onDidChangeState(
+            "debugging.dbconnect",
+            async (state) => {
+                if (state.avaliable) {
+                    window.showInformationMessage("Db Connect is enabled");
```
[the remark this answers is not in the document]
Comment (kartikgupta-db) [reply packages/databricks-vscode/src/extension.ts:394]: yep. this is a placeholder

File: packages/databricks-vscode/src/language/ConfigureAutocomplete.ts [hunk 3]
```diff
@@ -220,9 +163,13 @@ export class ConfigureAutocomplete implements Disposable {
         }
 
         //TODO: Make sure that pyspark is not updated if it is already installed
-        const execCommand = execCommandParts
-            .concat(["-m", "pip", "install", "pyspark"])
-            .join(" ");
+        const execCommand = [
```
[the remark this answers is not in the document]
Comment (kartikgupta-db) [reply packages/databricks-vscode/src/language/ConfigureAutocomplete.ts:166]: On looking through the extension, there doesn't seem to exists a command for this.

**Review by fjakobs — APPROVED** [review]
Screenshot or video would be nice to see it in action but otherwise looks good to me.

**Pull request merged**